libcontainer/rootfs_linux.go

```diff
@@ -152,14 +153,46 @@ func mountToRootfs(m *configs.Mount, rootfs, mountLabel string) error {
 		}
 		return nil
 	case "tmpfs":
+		copyUp := m.Extensions&configs.EXT_COPYUP == configs.EXT_COPYUP
+		tmpDir := ""
 		stat, err := os.Stat(dest)
 		if err != nil {
 			if err := os.MkdirAll(dest, 0755); err != nil {
 				return err
 			}
 		}
-		if err := mountPropagate(m, rootfs, mountLabel); err != nil {
-			return err
+		if copyUp {
+			tmpDirPrefix := os.Getenv("XDG_RUNTIME_DIR")
```

Comment:
Yeah we can't leak it from outside with this code so was thinking whether setting in the config might be an acceptable compromise. Else we can special case leak it.

Sent from my iPhone

> On May 27, 2016, at 3:15 PM, Michael Crosby <notifications@github.com> wrote:
> 
> In libcontainer/rootfs_linux.go:
> 
> >  		stat, err := os.Stat(dest)
> >  		if err != nil {
> >  			if err := os.MkdirAll(dest, 0755); err != nil {
> >  				return err
> >  			}
> >  		}
> > -		if err := mountPropagate(m, rootfs, mountLabel); err != nil {
> > -			return err
> > +		if copyUp {
> > +			tmpDirPrefix := os.Getenv("XDG_RUNTIME_DIR")
> I don't see how this would ever be set inside the container's init.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>